gap_hci.go

```diff
@@ -420,3 +441,51 @@ func (a *Advertisement) Start() error {
 func (a *Advertisement) Stop() error {
 	return a.adapter.hci.leSetAdvertiseEnable(false)
 }
+
+// SetServiceData sets the service data for the advertisement.
```

Comment:
Here are some points I think it might be helpful to cover in the method documentation here, to make it more clear how `SetServiceData` needs to be used and what its effects are:

1. It causes a new advertisement to be broadcast, with the updated service data that has been passed in.
2. Just updating the data in the `[]ServiceDataElement` that has already passed in to `Configure` via `AdvertisementOptions.ServiceData` will not change the data being broadcast in the advertisement. `SetServiceData` must be called for the new data to go into effect.
3. What happens if `Configure` or `Start` have not yet been called on the `Advertisement`?
4. What effect does this have if the `Advertisement` has been configured with the `Interval` option?